thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java

```diff
@@ -8,39 +8,39 @@
 public class AnomalyResultDAO extends AbstractJpaDAO<AnomalyResult> {
 
   private static final String FIND_BY_COLLECTION_TIME =
-      "SELECT r FROM AnomalyResult r WHERE r.collection = :collection "
+      "SELECT r FROM AnomalyResult r WHERE r.function.collection = :collection "
           + "AND ((r.startTimeUtc >= :startTimeUtc AND r.startTimeUtc <= :endTimeUtc) "
           + "OR (r.endTimeUtc >= :startTimeUtc AND r.endTimeUtc <= :endTimeUtc))";
 
   private static final String FIND_BY_COLLECTION_TIME_METRIC =
-      "SELECT r FROM AnomalyResult r WHERE r.collection = :collection AND r.metric = :metric "
+      "SELECT r FROM AnomalyResult r WHERE r.function.collection = :collection AND r.function.metric = :metric "
           + "AND ((r.startTimeUtc >= :startTimeUtc AND r.startTimeUtc <= :endTimeUtc) "
           + "OR (r.endTimeUtc >= :startTimeUtc AND r.endTimeUtc <= :endTimeUtc))";
 
   private static final String FIND_BY_COLLECTION_TIME_METRIC_DIMENSION =
-      "SELECT r FROM AnomalyResult r WHERE r.collection = :collection" + "AND r.metric = :metric "
-          + "AND r.dimensions IN :dimensions "
+      "SELECT r FROM AnomalyResult r WHERE r.function.collection = :collection" + "AND r.function.metric = :metric "
+          + "AND r.function.dimensions IN :dimensions "
```

Comment:
AnomalyFunction.dimension is not the same as AnomalyResults.dimensions. In anomaly function, dimensions stores the dimensions to use for drill down (eg: countryCode), whereas in anomaly results, this value is the actual dimension key (eg: (*,*,*,norway,*)